lua/lean/_util.lua

```diff
@@ -12,6 +12,20 @@ function M.tbl_repeat(value, times)
   return result
 end
 
+--- Fetch the diagnostics for all Lean LSP clients from the current buffer.
+function M.lean_lsp_diagnostics(opts, bufnr)
+  bufnr = bufnr or 0
+  opts = opts or {}
+  local diagnostics = {}
+  for _, client in pairs(vim.lsp.buf_get_clients(bufnr or 0)) do
+    if client.name:match('^lean') then
+      opts.namespace = vim.lsp.diagnostic.get_namespace(client.id)
+      vim.list_extend(diagnostics, vim.diagnostic.get(0, opts))
```

Comment:
```suggestion
      vim.list_extend(diagnostics, vim.diagnostic.get(bufnr, opts))
```

Reply:
Nice catch, I'd just noticed this myself, fixing, but first fighting with trying to get `vim.tbl_map` to do what I want.

Reply:
(Spoiler: I gave up.)

Reply:
Huh, what exactly was the issue?

Reply:
I couldn't get the tests to pass after changing to:

```lua
function M.lean_lsp_diagnostics(opts, bufnr)
  bufnr = bufnr or 0
  opts = opts or {}
  return vim.tbl_flatten(vim.tbl_map(function(client)
    if not client.name:match('^lean') then return {} end
    opts.namespace = vim.lsp.diagnostic.get_namespace(client.id)
    return vim.diagnostic.get(bufnr, opts)
  end, vim.lsp.buf_get_clients(bufnr)))
end
```

and got fed up fighting with Lua.